pkg/ebpf/state.go

```diff
@@ -89,14 +98,18 @@ func NewDefaultNetworkState() NetworkState {
 
 // NewNetworkState creates a new network state
 func NewNetworkState(clientExpiry time.Duration, maxClosedConns, maxClientStats int) NetworkState {
-	return &networkState{
+	ns := &networkState{
 		clients:        map[string]*client{},
 		telemetry:      telemetry{},
 		clientExpiry:   clientExpiry,
 		maxClosedConns: maxClosedConns,
 		maxClientStats: maxClientStats,
 		buf:            &bytes.Buffer{},
 	}
+
+	go ns.expvarStats()
```

Comment:
Instead of having all these goroutines in each component, can we just have one in ebpf/tracer.go that regularly grabs the data from each component and forwards it?

Reply:
Initially I thought maybe there are too many data types for metrics and I don't want to create a giant switch to handle each type, but now I realized that all of them could be int64, I think this could be done.